gguf-py/gguf/tensor_mapping.py

```diff
@@ -393,12 +451,16 @@ def __init__(self, arch: MODEL_ARCH, n_blocks: int):
             self.mapping[tensor_name] = (tensor, tensor_name)
             for key in keys:
                 self.mapping[key] = (tensor, tensor_name)
+        if arch in self.arch_block_mappings_cfg:
+            block_mappings = self.arch_block_mappings_cfg[arch]
```

Comment:
So the idea is to keep only "conflicting" block mappings in architecture-specific mappings and "non-conflicting" mappings in general mappings? I think using dict.update() is a better idea then. Mappings for ARCTIC arch would be shortened to:

```
    # architecture-specific block mappings
    arch_block_mappings_cfg: dict[MODEL_ARCH, dict[MODEL_TENSOR, tuple[str, ...]]] = {
        MODEL_ARCH.ARCTIC: {
            MODEL_TENSOR.FFN_NORM: (
                "model.layers.{bid}.residual_layernorm",
            ),
            MODEL_TENSOR.FFN_NORM_EXP: (
                "model.layers.{bid}.post_attention_layernorm",
            ),
        },
    }
```
while in the TensorNameMap init we would only have to add:
```
        if arch in self.arch_block_mappings_cfg:
            self.block_mappings_cfg.update(self.arch_block_mappings_cfg[arch])
```
What do you think?

Reply:
> So the idea is to keep only "conflicting" block mappings in architecture-specific mappings and "non-conflicting" mappings in general mappings?

Yes, exactly.

> What do you think?

I think using `dict.update()` would be good. My proposed approach would have made a copy of the dict, but you're right, updating in-place would work too and would be better, since the original `block_mappings_cfg` isn't used later on (I think?).

I agree with using `dict.update()` for this.

Reply:
OK, done